test/context.js

```diff
@@ -8,6 +8,8 @@ exports = module.exports = function(req, res){
   var socket = new Stream.Duplex();
   req = req || { headers: {}, socket: socket, __proto__: Stream.Readable.prototype };
   res = res || { _headers: {}, socket: socket, __proto__: Stream.Writable.prototype };
+  req.socket = req.socket || socket;
+  res.socket = res.socket || socket;
```

Comment:
What exactly does this do? `res.socket` is already defined as `socket` 2 lines up, right?